src/OrchardCore.Modules/OrchardCore.Liquid/Services/LiquidTemplateManager.cs

```diff
@@ -72,11 +72,9 @@ public async Task<IHtmlContent> RenderHtmlContentAsync(string source, TextEncode
                 }
             }
 
-            var htmlContentWriter = new ViewBufferTextWriterContent();
-
+            using var htmlContentWriter = new ViewBufferTextWriterContent();
             await result.RenderAsync(htmlContentWriter, encoder, context, model);
-
-            return htmlContentWriter;
+            return new HtmlString(htmlContentWriter.ToString());
```

Comment:
Sorry, as usual i didn't check it in details as i am working on other things ;)

But even if it is just a feeling, i prefer to point it out when i'm not sure

Here looks like we allocate a string in place of returning our `ViewBufferTextWriterContent` and let it write to the final output later, this through its `WriteTo()` as it is an `IHtmlContent`, and by the mvc `ViewBufferTextWriter`.

Reply:
@jtkech the PR is closed now, but check this instead https://github.com/OrchardCMS/OrchardCore/issues/9209

Reply:
Okay